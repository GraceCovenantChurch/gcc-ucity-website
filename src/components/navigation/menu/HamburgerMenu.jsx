import React from "react";
import PropTypes from "prop-types";

import makeStyles from '@mui/styles/makeStyles';
import IconButton from "@mui/material/IconButton";
import Drawer from "@mui/material/Drawer";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemText from "@mui/material/ListItemText";

import MenuIcon from "@mui/icons-material/Menu";
import CloseIcon from "@mui/icons-material/Close";

import DrawerLinks from "./drawer/DrawerLinks";

import Link from "components/link/Link";
import imageURLMobile from "static/gcclogo-black.png";

import { grey } from '@mui/material/colors';

const DRAWER_ANCHOR = "top";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
    color: grey[900],
  },
  menuText: {
    textAlign: "left",
  },
  title: {
    flexGrow: 1,
  },
  logo: {
    height: 80,
  },
  logoContainer: {
    flex: 1,
  },
  close: {
    float: "left",
    marginRight: "auto",
  },
  headerBar: {
    display: "flex",
    justifyContent: "flex-end",
    marginRight: theme.spacing(2),
    marginLeft: theme.spacing(2),
  },
}));

const HamburgerMenu = (props) => {
  let menuData = props.menu;
  const classes = useStyles();
  const [drawer, setDrawer] = React.useState(false);

  const toggleDrawer = (open) => (event) => {
    if (
      event.type === "keydown" &&
      (event.key === "Tab" || event.key === "Shift")
    ) {
      return;
    }

    setDrawer(open);
  };

  return (
    <React.Fragment>
      <IconButton
        edge="start"
        className={classes.menuButton}
        color="inherit"
        aria-label="menu"
        onClick={toggleDrawer(true)}
        size="large">
        <MenuIcon />
      </IconButton>
      <Drawer
        anchor={DRAWER_ANCHOR}
        open={drawer}
        onClose={toggleDrawer(false)}
      >
        <List>
          <List className={classes.headerBar}>
            <ListItem className={classes.logoContainer}>
              <img
                alt="nav-logo"
                className={classes.logo}
                src={imageURLMobile}
              />
            </ListItem>
            <IconButton
              edge="end"
              className={classes.close}
              color="inherit"
              aria-label="menu"
              onClick={toggleDrawer(false)}
              size="large">
              <CloseIcon />
            </IconButton>
          </List>
          <ListItem
            button
            component={Link}
            to={"/"}
            onClick={toggleDrawer(false)}
            onKeyDown={toggleDrawer(false)}
          >
            <ListItemText className={classes.menuText} primary={"Home"} />
          </ListItem>
          <DrawerLinks
            toggleDrawer={(flag) => toggleDrawer(flag)}
            data={menuData}
          />
        </List>
      </Drawer>
    </React.Fragment>
  );
};

HamburgerMenu.propTypes = {
  menu: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      children: PropTypes.any,
    })
  ),
};

export default HamburgerMenu;
