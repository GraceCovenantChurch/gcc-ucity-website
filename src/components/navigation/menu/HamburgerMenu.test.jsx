import React from "react";
import "../../../__mocks__/matchMedia";
import { Context as ResponsiveContext } from "react-responsive";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import HamburgerMenu from "./HamburgerMenu";

const menuOptions = [
  {
    title: "About",
    type: "dropdown",
    children: [{ title: "Welcome", type: "link", children: "/welcome" }],
  },
  { title: "Family Groups", type: "link", children: "/familygroup" },
];

describe("HamburgerMenu", () => {
  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 300 }}>
          <HamburgerMenu menu={menuOptions} />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(mobile).toMatchSnapshot();

    const { container: desktop } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 1000 }}>
          <HamburgerMenu menu={menuOptions} />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(desktop).toMatchSnapshot();
  });
});
