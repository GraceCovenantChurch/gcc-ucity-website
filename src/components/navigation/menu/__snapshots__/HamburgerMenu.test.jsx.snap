// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu matches the snapshot 1`] = `
<div>
  <button
    aria-label="menu"
    class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`HamburgerMenu matches the snapshot 2`] = `
<div>
  <button
    aria-label="menu"
    class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
